library/common/common/lambda_logger_delegate.cc

```diff
@@ -0,0 +1,29 @@
+#include "library/common/common/lambda_logger_delegate.h"
+
+#include <iostream>
+
+#include "common/common/lock_guard.h"
+
+namespace Envoy {
+namespace Logger {
+
+LambdaDelegate::LambdaDelegate(LogCb log_callback, FlushCb flush_callback,
+                               DelegatingLogSinkSharedPtr log_sink)
+    : SinkDelegate(log_sink), log_callback_(log_callback), flush_callback_(flush_callback) {
+  setDelegate();
+}
+
+LambdaDelegate::~LambdaDelegate() { restoreDelegate(); }
+
+void LambdaDelegate::log(absl::string_view msg) {
+  Thread::LockGuard lock(mutex_);
```

Comment:
What's the purpose for this mutex?

Reply:
If the delegate is used in a multi-threaded environment log calls could come from multiple threads. This is also an artifact of having written this class originally in Envoy. I can remove this given that Envoy Mobile is (for now) single threaded.